src/components/AssignmentSummary.jsx

```diff
@@ -106,6 +107,77 @@ export class AssignmentSummary extends Component {
     }
   }
 
+  calcHexBrightness(color) {
+    let hexcolor = color.replace("#", "");
+    if (hexcolor.length === 3) {
+      hexcolor = hexcolor.replace(/(.)/g, "$1$1");
+    }
+    const r = parseInt(hexcolor.substr(0, 2), 16);
+    const g = parseInt(hexcolor.substr(2, 2), 16);
+    const b = parseInt(hexcolor.substr(4, 2), 16);
+
+    const brightness = Math.floor((r * 299 + g * 587 + b * 114) / 1000);
+    const threshhold = 130;
+
+    const contrastingColor =
+      brightness > threshhold ? theme.colors.darkGray : theme.colors.white;
+    return contrastingColor;
+  }
+
+  calcRgbBrightness(color) {
+    const colorValuesArray = color
+      .slice(color.indexOf("(") + 1, color.indexOf(")"))
+      .split(",");
+
+    const r = parseInt(colorValuesArray[0]);
+    const g = parseInt(colorValuesArray[1]);
+    const b = parseInt(colorValuesArray[2]);
+    const alpha = colorValuesArray[3]
+      ? parseInt(colorValuesArray[3] * 100)
+      : 100;
+
+    const brightness = Math.floor((r * 299 + g * 587 + b * 114) / 1000);
```

Comment:
Same as the above comment. Makes me wonder if it would make sense to combine these two functions into one with a switch case so that the variables would only need to be declared once? But that is totally up to you :)